android/app/src/main/java/org/electroncash/electroncash3/Main.kt

```diff
@@ -493,6 +505,56 @@ class ChangePasswordDialog : AlertDialogFragment() {
 }
 
 
+class WalletRenameDialog : AlertDialogFragment() {
+    override fun onBuildDialog(builder: AlertDialog.Builder) {
+        builder.setTitle(R.string.Rename_wallet)
+                .setView(R.layout.wallet_rename)
+                .setPositiveButton(android.R.string.ok, null)
+                .setNegativeButton(android.R.string.cancel, null)
+    }
+
+    override fun onShowDialog() {
+        val walletName = arguments!!.getString("walletName")!!
+        etWalletName.setText(walletName)
+        etWalletName.setSelection(etWalletName.getText().length)
```

Comment:
If you set the initial state of an editable view in onShowDialog like this, the user's edits will be overwritten when they rotate the screen. 

When you rotate the screen, all the Dialog and View objects are discarded and then recreated from the layout XML file. The only UI state which is automatically kept over the rotation is user-editable things like the text and cursor position in an EditText, or the position of a SeekBar. onShowDialog is called after this state has been restored, so you need to be careful not to overwrite it there.

There are a couple of ways to fix situations like this:
* You can initialize the text in onFirstShowDialog, which is a base class method I added for that purpose. This approach is used by TransactionDialog. Notice that it still adds all the listeners in onShowDialog, because after rotation the old listeners have been discarded.
* If you're taking the text from the dialog arguments, you can null the arguments after you've read them. This appeoach is used by SendDialog. However, it probably wouldn't work here, because you need to remember the original wallet name somehow.

It's very easy for rotation to break things in Android, so I recommend you test it throughly whenever you work on the user interface.

Reply:
Ah, thanks for explaining the issue so clearly.  I've applied the first of the two solutions you suggest and it works nicely.

I notice however that the onscreen keyboard disappears during rotation.  This seems to happen in other places throughout the app, so perhaps it's to be expected in Android apps?  I compared with the bitcoin.com wallet and Badger wallet and unhelpfully neither of these allow rotation at all.  A couple other random apps I checked do keep the keyboard up during rotation though.